addons/sale/controllers/portal.py

```diff
@@ -207,14 +211,21 @@ def portal_quote_accept(self, res_id, access_token=None, partner_name=None, sign
         if not order_sudo.has_to_be_paid():
             order_sudo.action_confirm()
 
-        order_sudo.signature = signature
-        order_sudo.signed_by = partner_name
+        order_sudo.signed_by = name
+        order_sudo.signed_on = fields.Datetime.now()
```

Comment:
Would not it be better to make only one call to write with all values ?

Reply:
I suppose it would be better, I also noticed the action_confirm was called before the write, even if the write failed, so I changed that too.